cocotb/decorators.py

```diff
@@ -437,10 +437,21 @@ class test(_py_compat.with_metaclass(_decorator_helper, coroutine)):
             value representing simulation timeout (not implemented).
         expect_fail (bool, optional):
             Don't mark the result as a failure if the test fails.
-        expect_error (bool, optional):
-            Don't mark the result as an error if an error is raised.
-            This is for cocotb internal regression use 
-            when a simulator error is expected.
+        expect_error (bool or exception type or tuple of exception types, optional):
+            If ``True``, consider this test passing if it raises `any` :class:`Exception`, and failing if it does not.
+            If given an exception type or tuple of exception types, catching `only` a listed exception type is considered passing.
```

Comment:
```suggestion
            If given an exception type or tuple of exception types, catching *only* a listed exception type is considered passing.
```

Backticks are for references.